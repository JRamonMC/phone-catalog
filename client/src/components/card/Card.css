.btn {
  border-radius: 44px;
  text-align: center;
  margin: 0 auto;
}

.card-cont {
  min-width: 240px;
  min-height: 350px;
  margin: 80px 0 40px 0;
  padding: 20px 4px;
  text-align: center;
  box-shadow: 0 15px 30px 0 rgba(0,0,0,.11), 0 5px 15px 0 rgba(0,0,0,.08);
  border-radius: 24px;
}

.card {
  transition: .3s transform cubic-bezier(.155,1.105,.295,1.12),.3s box-shadow,.3s -webkit-transform cubic-bezier(.155,1.105,.295,1.12);
  cursor: pointer;
}

.img {
  max-width: 200px;
}