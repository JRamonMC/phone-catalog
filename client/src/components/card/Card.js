import React from "react";
import { Link, useHistory } from "react-router-dom";
import "./Card.css";
import ShowImage from "../showImage/ShowImage";
import { deletePhoneFromCatalogue } from "../../api/api";

const Card = ({ phone, detail }) => {
  const history = useHistory();

  const deletePhone = (phone) => {
    const phoneId = phone._id;

    deletePhoneFromCatalogue(phoneId)
      .then((data) => {
        if (data.error) {
          console.log(data.error);
        } else {
          history.goBack();
        }
      })
      .catch((err) => {
        console.log(err);
      });
  };

  const extraDetail = (phone) => {
    return (
      <div>
        <p className="font-weight-light">Manufacturer: {phone.manufacturer}</p>
        <p className="font-weight-light">Color: {phone.color}</p>
        <p className="font-weight-light">Screen: {phone.screen}</p>
        <p className="font-weight-light">Processor: {phone.processor}</p>
        <p className="font-weight-light">RAM: {phone.ram}</p>
        <button
          className="btn btn-secondary"
          onClick={() => {
            deletePhone(phone);
          }}
        >
          Delete
        </button>
      </div>
    );
  };

  const moreDetail = () => {
    return (
      <div>
        <Link to={`/seephone/${phone._id}`}>
          <button className="btn btn-info mt-2"> More details </button>
        </Link>
      </div>
    );
  };

  return (
    <div className="card m-10 card-cont center">
      <div>
        <ShowImage className="img" item={phone} url=""></ShowImage>
        <h3>{phone.name}</h3>
        <p className="font-weight-light">Price: {phone.price} $</p>
        <p className="font-weight-light">{phone.description}</p>
        {detail === true ? extraDetail(phone) : moreDetail()}
      </div>
    </div>
  );
};

export default Card;
