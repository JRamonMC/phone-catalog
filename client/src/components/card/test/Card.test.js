import React from "react";
import renderer from "react-test-renderer";
import Card from "./../Card";
import { shallow } from "enzyme";

describe("Phone card", () => {
  const fakeDataPhone = {
    name: "test",
    manufacturer: "test-m",
    description: "test-d",
    color: "test-c",
    price: "1",
    screen: "test-s",
    processor: "2",
    ram: "3",
    imageRaw: "test-i",
  };


  it("Phone card without detail", () => {
	const wrapper = shallow(<Card phone={fakeDataPhone} detail={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Phone name on  card shows  properly", () => {
    const wrapper = shallow(<Card phone={fakeDataPhone} detail={false} />);
    expect(
      wrapper.containsAnyMatchingElements([<h3>test</h3>])
    ).toEqual(true);
  });

});
