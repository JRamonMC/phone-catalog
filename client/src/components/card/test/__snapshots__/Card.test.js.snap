// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phone card Phone card without detail 1`] = `
<div
  className="card m-10 card-cont center"
>
  <div>
    <ShowImage
      className="img"
      item={
        Object {
          "color": "test-c",
          "description": "test-d",
          "imageRaw": "test-i",
          "manufacturer": "test-m",
          "name": "test",
          "price": "1",
          "processor": "2",
          "ram": "3",
          "screen": "test-s",
        }
      }
      url=""
    />
    <h3>
      test
    </h3>
    <p
      className="font-weight-light"
    >
      Price: 
      1
       $
    </p>
    <p
      className="font-weight-light"
    >
      test-d
    </p>
    <div>
      <Link
        to="/seephone/undefined"
      >
        <button
          className="btn btn-info mt-2"
        >
           More details 
        </button>
      </Link>
    </div>
  </div>
</div>
`;
